h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamRowToRawDataConverter.java

```diff
@@ -0,0 +1,32 @@
+package hex.genmodel.algos.gam;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.easy.CategoricalEncoder;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.easy.RowToRawDataConverter;
+import hex.genmodel.easy.exception.PredictException;
+
+import java.util.Map;
+
+import static hex.genmodel.utils.ArrayUtils.nanArray;
+
+public class GamRowToRawDataConverter extends RowToRawDataConverter {
+  GamMojoModelBase _m;
+  public GamRowToRawDataConverter(GenModel m, Map<String, Integer> modelColumnNameToIndexMap, Map<Integer, CategoricalEncoder> domainMap, EasyPredictModelWrapper.ErrorConsumer errorConsumer, EasyPredictModelWrapper.Config config) {
+    super(m, modelColumnNameToIndexMap, domainMap, errorConsumer, config);
+    _m = (GamMojoModelBase) m;
+  }
+  
+  @Override
+  public double[] convert(RowData data, double[] rawData) throws PredictException {
+    if (data.containsKey(_m._gamColNamesCenter[0][0])) {
+      return super.convert(data, rawData);  // case where the input already contains the expanded gam columns
+    } else {
+      rawData = nanArray(_m.getTotFeatureSize());
+      rawData = super.convert(data, rawData);
```

Comment:
The `rawData = super.convert(data, rawData);` part fills all the columns but the gamified ones, and the `_m.addExpandGamCols(rawData, data);` should only fill the GAM coefs. Insteada, the latter somehow erases the original features and puts the gam coefficients at the beginning of the array while leaving everything else as NaN.

![Screenshot from 2020-07-16 13-46-47](https://user-images.githubusercontent.com/8769110/87669294-05736500-c76e-11ea-918a-d11792211327.png)

![Screenshot from 2020-07-16 13-47-00](https://user-images.githubusercontent.com/8769110/87669304-0906ec00-c76e-11ea-846e-181f16cc3b9c.png)


The screenshots above demonstrate this behavior.

Reply:
Fixed.  I hope.